.github/workflows/deploy.yml

```diff
@@ -0,0 +1,60 @@
+name: Deploy
+
+permissions:
+  id-token: write
```

Comment:
Only `deploy` job requires this permission so it should be moved under it and not global.